### ISSUE TITLE
Immutables / javax.validation not working since new BV spec (2.0)

### ISSUE THREAD

**Issue by asereda-gs** [immutables/immutables#641]
Hello,

I'm not sure if this a problem of Immutables ([Gson](https://immutables.github.io/json.html#gson) / [Repository](https://immutables.github.io/mongo.html)) or [Bean Validation 2.0](http://beanvalidation.org/news/2017/05/24/bean-validation-2-0-public-review-update/) but seems they don't work together (since 2.0) in the following example: 

```java
import javax.validation.constraints.Size;
...

/**
 * An example which fails. Using javax.validation 1.1.0 - works. 
  * 2.0.0.Beta2 doesn't work
 */
@Value.Immutable
@Gson.TypeAdapters // removing this annotation makes things work
@Mongo.Repository // similar to @Gson (will trigger compilation error)
public interface Entity {
    @Size(max = 10) // removing this also works
    String name();
}
```
### Error
```
Error:java: org.immutables.value.internal.$processor$.$Processor threw java.lang.IllegalArgumentException: Cannot parse type from input string 'java.lang.@javax.validation.constraints.Size(max = 10) String'. unexpected trailing terms 'max=10)String'
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser.doParse($Type.java:870)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser.parse($Type.java:715)
  	at org.immutables.value.internal.$processor$.meta.$GsonTypeTokens.parseType($GsonTypeTokens.java:39)

```

Bean Validation: 2.0.0.Beta2 (v1.1.0 - works fine)
Immutables: 2.5.3

**Comment by elucash** [comment]
Probably because they switched annotations to be TYPE_USE. And we're having problem parsing it in some places (like gson analyzer), but not generally. The problem can be solved by improving parsing routine to support type annotations (in qualified form like `java.lang.@some.Ann String`)

**Comment by asereda-gs** [comment]
You mean [Type.java](https://github.com/immutables/immutables/blob/master/value-processor/src/org/immutables/value/processor/encode/Type.java) ?

**Comment by elucash** [comment]
I'll make a fix, I was going to do this before, but found easier workaround last time, but it has to be fixes anyway.

**Comment by elucash** [comment]
was released in 2.5.4 thank you!

**Comment by archieby** [comment]
Hi, not sure if it the right place to post it, but for me, the example provided above generates the following immutable class:

`private final java.lang.@Size(max = 10) String name;`

Please see the **java.lang.** garbage prefixing the `@Size` annotation.
With it there - validation does not work as expected
With it manually removed - everything works fine.
What might be the reason for this behavior?
2.5.4-2.5.6 versions checked
